### ISSUE TITLE
Redirection placed before subshell triggers SC1036

### ISSUE THREAD

**Issue by ncoop** [koalaman/shellcheck#716]
```shell
#!/bin/sh

acommand >somewhere
>somewhere acommand
(commands) >somewhere
>somewhere (commands)
```
First three are OK.
Last one throws `'(' is invalid here. Did you forget to escape it? [SC1036]`

_edit_
Maybe this is related:
```shell
time acommand
time (commands)
```
The second triggers `Trying to declare parameters? Don't. Use () and refer to params as $1, $2.. [SC1065]`

Seems to be a problem with recognizing subshells that open after the first token.

_edit_
Shellcheck.net (newer than my 0.3.7) knows about the time example, so that's good.

**Comment by kurahaupo** [comment]
On Sun, 24 Jul 2016, ncoop wrote:
> >somewhere (commands)
> ```
> First three are OK.
> Last one throws `'(' is invalid here. Did you forget to escape it? [SC1036]`

Shellcheck is correct; that last one is not valid shell syntax.

The rules for placement of redirections on compound statements are rather
more stringent that for simple statements; basically they must always go
after the closing token: ")", "))", "]]", "}", "done", "esac", or "fi".

There are also complications is you use aliases, as they are expanded before
compound statements are recognized (and hence, before redirections).

-Martin

**Comment by ncoop** [comment]
TIL redirections must always go after the closing token. :bulb:

It'd be good if shellcheck explains that.

**Comment by koalaman** [comment]
I investigated how common this is, but I wasn't able to find any other examples, and a test fix mostly showed more confusing error messages for nonsensical code like `a=expr (1+2)` which was then parsed as a subshell with a prefix assignment.

I'll just leave this as is.
